tools/isobuild/package-api.js

```diff
@@ -339,13 +339,39 @@ _.extend(PackageAPI.prototype, {
     var errors = [];
     _.each(paths, function (path) {
       forAllMatchingArchs(arch, function (a) {
+        const isAsset = !! (fileOptions && fileOptions.isAsset);
+
+        // Check if we have already added a file at this path
         if (_.has(self.sources[a], path)) {
-          errors.push("Duplicate source file: " + path);
+          const otherSource = self.sources[a][path];
+
+          // Files can actually be added twice for each architecture: once as
+          // an asset, and once as a source file
+          const otherFileOptions = otherSource.fileOptions;
+          const otherFileIsAsset = !! (otherFileOptions &&
```

Comment:
Ohh, I guess you can still have two calls to `addFiles` for the same file, but only one per arch. Hmm, I guess `isAsset: "both"` could be a good option as well. Greenspan expressed an opinion in favor of the other API; @dgreensp what do you think?